components/elm/src/external_models/emi/src/em/alquimia/ExternalModelAlquimiaMod.F90

```diff
@@ -0,0 +1,3019 @@
+        mineral_specific_surface_area_tmp,&
+        surface_site_density_tmp,&
+        cation_exchange_capacity_tmp,&
+        aux_doubles_tmp,&
+        aux_ints_tmp) 
+      
+    else ! Solve did not converge. Cut timestep, and bail out if too short
+      if(actual_dt/2 < min_dt) then
+        call c_f_string_ptr(this%chem_status%message,status_message)
+        ! Sometimes solve fails because time step is too short (not sure why)
+        ! I wonder if at bailout we should first try pausing transport and solving each layer separately?
+        write(msg,'(a,i3,a,f5.3,a,i4,a,i3,a,i5)') "Error: Alquimia ReactionStepOperatorSplit failed to converge after ",num_cuts," cuts to dt = ",actual_dt,' s. Newton iterations = ',this%chem_status%num_newton_iterations,' Layer = ',j!," Col = ",c
+        call print_alquimia_state(this)
+        write(iulog,*),'Alquimia status message: '//status_message
+        call endrun(msg=msg)
+      else
+        exit ! Drop out of the layer loop to start over at shorter time step
+      endif
+    endif
+    enddo ! Layer loop
+
+
+    if(actual_dt<=60.0_r8) then
+      ! write(iulog,*),'Alquimia: Time step cut to 60 s. Attempting to solve by pausing transport and solving layer by layer'
+      do j=1,nlevdecomp
+            ! Update properties from ELM
+        this%chem_state%porosity =    porosity(j)
+        this%chem_state%temperature = temperature(j) - 273.15
+        this%chem_properties%volume = volume(j)
+        this%chem_properties%saturation = sat(j) ! Set minimum saturation to stop concentrations from blowing up at low soil moisture
+        call this%copy_ELM_to_Alquimia(j,water_density,&
+                                          aqueous_pressure,&
+                                          total_mobile,&
+                                          total_immobile,&
+                                          mineral_volume_fraction,&
+                                          mineral_specific_surface_area,&
+                                          surface_site_density,&
+                                          cation_exchange_capacity,&
+                                          aux_doubles,&
+                                          aux_ints) 
+        call run_onestep(this,dt,num_cuts,ncuts)
+        call this%copy_Alquimia_to_ELM(j,water_density_tmp,&
+                                        aqueous_pressure_tmp,&
+                                        total_mobile_tmp,free_mobile_tmp,&
+                                        total_immobile_tmp,&
+                                        mineral_volume_fraction_tmp,&
+                                        mineral_specific_surface_area_tmp,&
+                                        surface_site_density_tmp,&
+                                        cation_exchange_capacity_tmp,&
+                                        aux_doubles_tmp,&
+                                        aux_ints_tmp)
+      enddo
+    endif
```

Comment:
I think we should remove this block. I'm not sure it's in the right place and it's usually not necessary